### ISSUE TITLE
1.5: Portability issues with new audio scheme

### ISSUE THREAD

**Issue by RC-SOAR** [FrSkyRC/ETHOS-Feedback-Community#3584]
Love the features in 1.5, but the new Audio scheme makes model setups less portable - a problem when for 3rd party template creators and people who want to swap setups with others.

This is primarily because of the new 'voices' and - in particular - the voice dependencies in SFs.  

People only speak one language, so why not pick up the system files automatically? And while it's nice to have multiple folders for custom sounds, it shouldn't be at the expense of portability - a single 'custom' folder is IMO sufficient for most users.

So two suggestions:

- Pick up system sounds from the {lang/sublang} option in the System>General tab. 
- Pick up custom sounds from a standard 'Custom' folder (automatically generated if not already present)

_This would remove the need for 'voices', which means the Voice option can be removed from SF's._

The would result in setups which are more portable.

**Comment by 4qr1hu9n7** [comment]
Works just fine. Please don't change it. Very simple to use.

**Comment by pstasek** [comment]
Voices have much more potential than just simple change of voice. This is one of the key features for me in 1.5. For example allows me to manage multiple IMAC sequence callings in a single global special function. By selecting a different voice folder I can fly different  sequence.

**Comment by RC-SOAR** [comment]
> Voices have much more potential than just simple change of voice. 

Okay, point well taken :) I also understand that to change things radically at this stage would not be possible. 

Here's another option which would increase portability without impacting existing setups:

- expand the SF's voice dropdown to include a special voice 'CustomVoice'
- CustomVoice would point to a *fixed* subfolder location regardless of tx type or model. For example '/audio/custom'. 

So that way template designers could set their custom SF's to 'CustomVoice'. Then instruct users to copy the custom files to the /audio/custom folder. 

This would make things simpler for users and template designers (and anyone who wants to share their setup). And there would no impact on existing 1.5 setups.

**Comment by strgaltdel** [comment]
I had a similar idea, but then I asked myself
"and how do you handle language specific custom voices" ? 

maybe the English don't want the wav's from a published German template,
and the French don't want to hear them from the English template,
and.....

so we would end up in another subfolder 
and we are close to our current "starting position",

That was the point at which I abandoned further considerations



now we have three (language specific)  different folder in parallel
- the default
- user#1
- user#2

my "personal best practise" would be
- default for (may be generated by a tts system) all these standard system files which arrive with ethos
- user1 for custom audio files
- user2 for sounds like individual alarms, chimes and so on

so a template would get his custom sounds from in the system settings configured folder "user1" whatever the folder is named
This would require a "custom sound subfolder" in each lang specific folder
AND
the definition of this user specific folder in the system settings

i do presume that each user who wants to handle with own, specific  and additional sound files will have a motivation to think about the "rules" of the new concept

**Comment by RC-SOAR** [comment]
> "and how do you handle language specific custom voices" ?

The template _writer_ can supply multiple language files in the package as normal. 

The _user_  only needs files for one languge, and they can simply copy these into the CustomVoice folder. 

audio/UNIVERSAL_CUSTOM

> i do presume that each user who wants to handle with own, specific and additional sound files will have a motivation to think about the "rules" of the new concept

Many users will struggle with the new rules, especially those who use canned setups (they use them precisely because they should be easy to configure). This measure is to support those users, and to make it easier for users who want to swap setups. 

Also it'll make it easier for template writers, since all SF's can point to 'CustomVoice'.

(No impact on existing 1.5 setups, people can still split up their sound files into voice1, voice2 and voice3 if they wish, bearing in mind that those voice folders only apply to their own device).

**Comment by mawzthefinn** [comment]
Maybe a Global option, with sounds in the 1.4 location (/Audio)

User can select Global, or Voice 1, 2 & 3 for Play Track. Templates use Global (and this solves some of the custom audio issues users are having in the 1.5 transition, since Global is just the old location)

**Comment by strgaltdel** [comment]
so we are discussing if a 
audio/lang/custom

or a 
audio/custom/(lang?) should be used.

i do undestood the wish to divide between
 "individual user" = audio/lang

and a "3rd party" / template sources
"audio/custom/(lang?)"

structure


what would end in at least one more folder mentioned above

all in all:
what would you prefer: 

- a single "one language" folder ("audio/global") would come into (my) mind

- or a folder with a language specific substructure, so that changing system voice would also apply to this structure?

oh, mawz had "quicker fingers",
so one more "global" voice..

**Comment by RC-SOAR** [comment]
> so we are discussing if a
> audio/lang/custom [...]

The extra voice + folder must be independent of lang, and  tx. 
In terms of naming, mawz's suggestion of 'global' instead of 'custom'  makes sense. So:

```
folder = /audio/global, or just /audio
voice = 'voiceGlobal' (system supplied voice)
```

Typical use case: 

1. Template author places provides custom .wav files (en, it, de etc.) in folders in zip file.  Template has Play Audio SF's pointing to 'voiceGlobal'
2. Template user copies .wav files - for his language only - into /audio/global. No changes necessary to the template or other folders.

The rest of the structure would be as now, no impact on existing setups or other folders. Users can continue using voice1, voice2 etc.

**Comment by bsongis-frsky** [comment]
I didn't think it would bring so many questions / frustration on forums. 

Don't forget that languages are not only a set of audio files, there is a grammar associated to each language. For example in CZ you have 3 forms of plural ...

I would like not to do another change, which will perhaps bring more questions for users who have done it right.

I propose a very simple change in 1.5.2

1) create on all languages a "default" directory. i.e. [en/us, en/gb, en/default], [fr/male, fr/female, fr/default], etc.

2) if the system audio voice1 (main voice) has not already being set, write "default" there, as we are sure that the directory exists

This way with the next update (audio files + firmware), everybody should have audio working

**Comment by bsongis-frsky** [comment]
Ready for 1.5.2
